plugins/github.js

```diff
@@ -10,24 +10,35 @@ exports.init = function(config, cimpler) {
          return next();
       }
 
-      try {
-         var build = extractBuildInfo(req.body);
-         if (build) {
+      const payload = JSON.parse(req.body.payload);
+      const event = req.headers['X-GitHub-Event'];
+      var build = null;
+
+      /**
+       * The webhook sent must be a push event OR a pull_request event,
+       * not both.
+       */
+      if (event == 'push') {
+         build = extractPushBuildInfo(payload);
+      } else if (event == 'pull_request') {
+         build = extractPullRequestBuildInfo(payload);
```

Comment:
I was going to say that maybe we should just modify the `extractBuildInfo` function to check for both of these events but im fine with the clear separation

Reply:
I think I like the separation.  It also keeps the functions a bit smaller.  It also makes adding different events easier if we want in the future.  Just add a different event to the switch.